_scripts/mime-db-shrinking-loader.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * electron-context-menu only needs mime-db for its save as feature.
+ * As we only activate save image and save as image features, we can remove all other mimetypes,
+ * as they will never get used.
+ * Which results in quite a significant reduction in file size.
+ * @param {string} source
+ */
+module.exports = function (source) {
+  const original = JSON.parse(source)
+
+  const reduced = {}
+
+  for (const mimeType of Object.keys(original)) {
+    if (mimeType.startsWith('image/') && original[mimeType].extensions &&
+      (!mimeType.startsWith('image/x-') || mimeType === 'image/x-icon' || mimeType === 'image/x-ms-bmp') &&
+      (!mimeType.startsWith('image/vnd.') || mimeType === 'image/vnd.microsoft.icon')) {
+
+      // Only the extensions field is needed, see: https://github.com/kevva/ext-list/blob/v2.2.2/index.js
+      reduced[mimeType] = {
+        extensions: original[mimeType].extensions
+      }
+    }
```

Comment:
Just make it easier to read
Coz the logic seems to be
1. Add all `image/` MIME types not matching specific prefixes with extensions
2. Add specific MIME types
```suggestion
    if (!mimeType.startsWith('image/')) { continue }
    if (original[mimeType].extensions == null) { continue }

    let shouldBeAdded = false
    if (!mimeType.startsWith('image/x-') && !mimeType.startsWith('image/vnd.')) {
      shouldBeAdded = true
    } else {
      shouldBeAdded = mimeType === 'image/x-icon' ||
        mimeType === 'image/x-ms-bmp' ||
        mimeType === 'image/vnd.microsoft.icon'
    }
    if (!shouldBeAdded) { continue }

    // Only the extensions field is needed, see: https://github.com/kevva/ext-list/blob/v2.2.2/index.js
    reduced[mimeType] = {
      extensions: original[mimeType].extensions
    }
```

Reply:
Current code is easier to read imo

Reply:
I agree with @efb4f5ff-1298-471a-8973-3d47447115dc I think the current code is more readable, also your code changes the logic, resulting in a different output.

I prefer the or/`||` operator over writing lots of single line ifs that have the same outcome, as to me it's a lot more readable, especially when the variable names are short.

Reply:
I agree with @efb4f5ff-1298-471a-8973-3d47447115dc I think the current code is more readable, also your code changes the logic, resulting in a different output.

I prefer the or/`||` operator over writing lots of single line ifs that have the same outcome, as to me it's a lot more readable, especially when the variable names are short.

Also we really don't need to check for `null` here, as this code runs on a json file with a known structure so the only options are non-existent/`undefined` or an array, so checking for loose equality with `null` is rather confusing. Also the truthy check already covers both `undefined` and `null`.